onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -196,8 +197,25 @@ Status LoadInitializerOrtFormat(const fbs::Tensor& fbs_tensor,
     const auto* fbs_raw_data = fbs_tensor.raw_data();
     ORT_RETURN_IF(nullptr == fbs_raw_data, "Missing raw data for initializer. Invalid ORT format model.");
 
-    // fbs_raw_data is uint8_t vector, so the size is byte size
-    initializer.set_raw_data(fbs_raw_data->Data(), fbs_raw_data->size());
+    if (can_use_flatbuffer_for_initializers && fbs_raw_data->size() > 127) {
+      initializer.set_data_location(ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL);
+
+      const void* data_offset = fbs_raw_data->Data();
+      auto offset = gsl::narrow<ExternalDataInfo::OFFSET_TYPE>(reinterpret_cast<uintptr_t>(data_offset));
+
+      ONNX_NAMESPACE::StringStringEntryProto* entry = initializer.mutable_external_data()->Add();
+      entry->set_key("location");
+      entry->set_value(ToUTF8String(onnxruntime::utils::kMemoryAddressTag));
```

Comment:
do we need to do anything else to ensure that a TensorProto with external data referring to a memory address doesn't get serialized to an ONNX model?

Reply:
I don't think we _need_ to. 

A user loading an ORT format model, providing the bytes directly, and setting additional flags to use that buffer for initializers only to later try and save the model back to protobuf format seems extremely unlikely and not worth the cost of protecting against. Worst case scenario if they did that they should get an error message about the invalid file path containing the special marker when attempting to load the resultant model which shouldn't take too long to figure out.